### PR TITLE
fix: sources: colombia: escape suppliers names

### LINKED ISSUE #258: Escape suppliers names in `_get_remote_award`
For suppliers' names such as "G & G INGENIEROS SAS" we need to escape the "&" character in the  `_get_remote_award` method. Otherwise, the endpoint recognizes the value as a parameter, e.g.:

https://www.datos.gov.co/resource/p6dx-8zbt.json?$where=id_del_portafolio='CO1.BDOS.5267263' AND nombre_del_proveedor='G & G INGENIEROS SAS'

returns:
```
{
  "error": true,
  "message": "Unrecognized arguments [ G INGENIEROS SAS]"
}
```

The endpoint returns the name of the supplier as "G  G INGENIEROS SAS"
https://www.datos.gov.co/resource/p6dx-8zbt.json?$where=id_del_portafolio='CO1.BDOS.5267263' AND nombre_del_proveedor='G  G INGENIEROS SAS'

That said, the strategy should be to remove the "&" character when querying that endpoint.

### PR DESCRIPTION
**Pull request by yolile**
closes #258

### CODE REVIEW

File: app/sources/colombia.py [hunk 1]
```diff
@@ -186,7 +188,7 @@ def get_borrower(borrower_identifier: str, documento_proveedor: str, entry: dict
     borrower_response_json = borrower_response.json()
     len_borrower_response_json = len(borrower_response_json)
 
-    if len_borrower_response_json > 1:
+    if len_borrower_response_json > 1 or len_borrower_response_json < 1:
```
Comment (jpmckinney) [inline app/sources/colombia.py:191]:
```suggestion
    if len_borrower_response_json != 1:
```

File: app/sources/colombia.py [hunk 2]
```diff
@@ -19,6 +19,8 @@
 
 
 def _get_remote_award(proceso_de_compra: str, proveedor_adjudicado: str) -> tuple[list[dict[str, str]], str]:
+    # To avoid '&' been taken as a query parameter
+    proveedor_adjudicado = proveedor_adjudicado.replace("&", "%26")
```
Comment (jpmckinney) [inline app/sources/colombia.py:23]: Shouldn't we escape all using urllib?

Comment (yolile) [reply app/sources/colombia.py:23]: I tried that but for some reason, it wasn't working, the API returns an empty response.

Comment (yolile) [reply app/sources/colombia.py:23]: Full context: the final URL with urllib is, for example: https://www.datos.gov.co/resource/p6dx-8zbt.json?$where=id_del_portafolio='CO1.BDOS.5572587' AND nombre_del_proveedor='David%20Alexander%20Rodr%C3%ADguez%20Cosme' if you try that URL in your browser it works, however, from the python code the API returns a empty array []. I'm not sure if we are encoding the URL again at some point and that is why it fails from the code.

Comment (yolile) [reply app/sources/colombia.py]: Indeed, that did the trick!

Comment (yolile) [reply app/sources/colombia.py]: Or not, according to the tests..

Comment (jpmckinney) [reply app/sources/colombia.py]: Ah, the single quotes need to be %27. Maybe `parse.quote` the entire query string?

Comment (yolile) [reply app/sources/colombia.py]: It works when I run the command manually so it is weird that is failing in the tests, but I will try that

Comment (yolile) [reply app/sources/colombia.py]: I think we need to mock up the _get_award method from that test anyway. So I will do both

**Pull request merged**